programs/more_decorator.py

```diff
@@ -24,18 +25,15 @@ def wrapper_function(*args, **kwargs):
     return wrapper_function
 
 
-counter = 0
-
-
 @retry
 def save_to_database(arg):
     print("Write to a database or make a network call or etc.")
     print("This will be automatically retried if exception is thrown.")
-    global counter
-    counter += 1
+
     # This will throw an exception in the first call
     # And will work fine in the second call (i.e. a retry)
-    if counter < 2:
+    if not hasattr(save_to_database, 'called'):
```

Comment:
I'm not sure this is a good change, it distracts from the example's focus on decorators?

Reply:
It is less line of code than the global counter (global = generally bad), so to me, it feels less distracting? But if you insist, I can revert that part of it. Let me know.

Reply:
@maxzheng Can you please check if we have already discussed `hasattr` in the book before this section? If yes, let's keep it. If no, can you please revert that part? Thanks!

Reply:
Interestingly there is no hasattr use. Reverted for now. It is a fairly useful function, I would suggest adding that in the future.